classes/helper.php

```diff
@@ -46,6 +46,21 @@ public static function scripttypeasstring(int $calltype): string {
             case profile::SCRIPTTYPE_WS:
                 return get_string('excimertype_ws', 'tool_excimer');
         }
+        throw new \coding_exception('Unsupported script type ' . $type);
+    }
+
+    /**
+     * @param int $type
+     * @return string
+     */
+    public static function logmethoddisplaytring(int $method): string {
+        switch ($method) {
+            case manager::LOGMETHOD_MANUAL:
+                return get_string('excimerlogmethod_manual', 'tool_excimer');
+            case manager::LOGMETHOD_AUTO:
+                return get_string('excimerlogmethod_auto', 'tool_excimer');
+        }
+        throw new \coding_exception('Unsupported logging method ' . $method);
```

Comment:
same again: if it is just display logic which isn't critical, I'd make this kind of code more forgiving and just print an 'unknown' instead of exploding

Reply:
Done